packages/slider/examples/controlled-audio-progress.example.js

```diff
@@ -0,0 +1,292 @@
+import React from "react";
+import "../styles.css";
+import {
+  Slider,
+  SliderHandle,
+  SliderTrack,
+  SliderTrackHighlight
+} from "../src";
+
+export const name = "Audio Progress";
+
+export const Example = () => {
+  const [value, setValue] = React.useState(0);
+  const [max, setMax] = React.useState(1000);
+  const [audio, audioState, audioControls, audioRef] = useAudio({
+    src: "http://techslides.com/demos/samples/sample.mp3",
+    autoPlay: false,
+    controls: true,
+    onLoadedMetadata: function(event) {
+      if (audioRef.current) {
+        setMax(Math.round(timeToMs(audioRef.current.duration)));
+      }
+    }
+  });
+
+  function handleChange(newValue) {
+    if (audioRef.current && audioControls) {
+      audioControls.seek(newValue / 1000);
+    }
+  }
+
+  React.useEffect(() => {
+    if (audioState.time != null) {
+      setValue(timeToMs(audioState.time));
+    }
+  }, [audioState.time]);
+
+  return (
+    <div>
+      <Slider
+        onChange={handleChange}
+        value={value}
+        min={0}
+        max={max}
+        getValueText={val => msToTime(val)}
+      >
+        <SliderTrack>
+          <SliderTrackHighlight />
+          <SliderHandle />
+        </SliderTrack>
+      </Slider>
+      {audio}
+    </div>
+  );
+};
+
+/// UTILS
+function msToTime(ms) {
+  let seconds = Math.floor((ms / 1000) % 60);
+  let minutes = Math.floor((ms / (1000 * 60)) % 60);
+  let hours = Math.floor((ms / (1000 * 60 * 60)) % 24);
+
+  hours = hours < 10 && hours > 0 ? "0" + hours : hours;
+  minutes = minutes < 10 ? "0" + minutes : minutes;
+  seconds = seconds < 10 ? "0" + seconds : seconds;
+
+  return (hours ? hours + ":" : "") + minutes + ":" + seconds;
+}
+
+function timeToMs(time) {
+  return Math.round(time * 1000);
+}
+
+// https://github.com/streamich/react-use/blob/master/src/util/createHTMLMediaHook.ts
+function useAudio(elOrProps) {
```

Comment:
Maybe put this in a separate file because it's so large?

Reply:
Moved to a new utils.js for examples.